### ISSUE TITLE
Installation of ansys-mapdl-core and ansys-mapdl-reader on macOS Monterey 12.4 (M1)

### ISSUE THREAD

**Issue by kp3393** [pyansys/pymapdl-reader#141]
### Before submitting the issue

- [X] I have searched among the existing issues
- [X] I am using a Python virtual environment

### Description of the bug

Hello team,
I'm having issues installing ansys-mapdl-core and ansys-mapdl-reader on macOS Monterey 12.4.
Here is the setup detail:
1. Using PyCharm 2022.1.2 (Professional Edition)
2. Using Conda Environment
3. Python version 3.9.5 

Please, find attached the details of the error while running "pip install ansys-mapdl-core" in 'ansys_mapdl_core.pdf' and for "pip install ansys-mapdl-reader" in 'ansys_mapdl_reader.pdf'

Summary of the errors:
Cannot build wheels for ansys-mapdl-reader

Building wheels for collected packages: ansys-mapdl-reader
Building wheel for ansys-mapdl-reader (setup.py) ... error
error: subprocess-exited-with-error
× python setup.py bdist_wheel did not run successfully.
│ exit code: 1

### Steps To Reproduce

1. Use conda virtual  environment
2. Try to install mapdl using "pip install ansys-mapdl-core" and "pip install ansys-mapdl-reader"

### Which Operating System are you using?

MacOS

### Which Python version are you using?

3.9

### Installed packages

(IRT) kaushal@Kaushals-Mini IRT % python -m pip freeze
absl-py==1.1.0
aiohttp @ file:///Users/runner/miniforge3/conda-bld/aiohttp_1649013207063/work
aiosignal @ file:///home/conda/feedstock_root/build_artifacts/aiosignal_1636093929600/work
ansys-api-mapdl==0.5.1
ansys-api-mapdl-v0==0.4.1
ansys-api-platform-instancemanagement==1.0.0b3
ansys-corba==0.1.0
ansys-dpf-core==0.4.2
ansys-dpf-post==0.2.2
ansys-grpc-dpf==0.4.0
ansys-platform-instancemanagement==1.0.2
appdirs==1.4.4
astunparse @ file:///home/conda/feedstock_root/build_artifacts/astunparse_1610696312422/work
async-timeout @ file:///home/conda/feedstock_root/build_artifacts/async-timeout_1640026696943/work
attrs @ file:///home/conda/feedstock_root/build_artifacts/attrs_1640799537051/work
autobahn==22.5.1
Automat==20.2.0
backports.functools-lru-cache @ file:///home/conda/feedstock_root/build_artifacts/backports.functools_lru_cache_1618230623929/work
bcrypt @ file:///Users/runner/miniforge3/conda-bld/bcrypt_1651441589427/work
blinker==1.4
brotlipy @ file:///Users/runner/miniforge3/conda-bld/brotlipy_1648854242877/work
cached-property @ file:///home/conda/feedstock_root/build_artifacts/cached_property_1615209429212/work
cachetools @ file:///home/conda/feedstock_root/build_artifacts/cachetools_1633010882559/work
certifi==2022.6.15
cffi @ file:///Users/runner/miniforge3/conda-bld/cffi_1636046173594/work
cftime @ file:///Users/runner/miniforge3/conda-bld/cftime_1649637183391/work
charset-normalizer @ file:///home/conda/feedstock_root/build_artifacts/charset-normalizer_1644853463426/work
click @ file:///Users/runner/miniforge3/conda-bld/click_1651215424788/work
colorama @ file:///home/conda/feedstock_root/build_artifacts/colorama_1655412516417/work
commonmark==0.9.1
constantly==15.1.0
cryptography @ file:///Users/runner/miniforge3/conda-bld/cryptography_1652967108255/work
cycler==0.11.0
Cython==0.29.30
dataclasses @ file:///home/conda/feedstock_root/build_artifacts/dataclasses_1628958434797/work
flatbuffers==1.12
fonttools==4.33.3
frozenlist @ file:///Users/runner/miniforge3/conda-bld/frozenlist_1648771983520/work
future @ file:///Users/runner/miniforge3/conda-bld/future_1649010215397/work
gast @ file:///home/conda/feedstock_root/build_artifacts/gast_1596839682936/work
geomdl==5.3.1
gmpy2 @ file:///Users/runner/miniforge3/conda-bld/gmpy2_1641732972152/work
google-api-core==2.8.2
google-api-python-client==2.51.0
google-auth @ file:///home/conda/feedstock_root/build_artifacts/google-auth_1629296548061/work
google-auth-httplib2==0.1.0
google-auth-oauthlib @ file:///home/conda/feedstock_root/build_artifacts/google-auth-oauthlib_1630497468950/work
google-pasta==0.2.0
googleapis-common-protos==1.56.2
grpcio @ file:///Users/runner/miniforge3/conda-bld/grpcio_1655262493691/work
h5py @ file:///Users/runner/miniforge3/conda-bld/h5py_1609497507927/work
httplib2==0.20.4
hyperlink==21.0.0
idna @ file:///home/conda/feedstock_root/build_artifacts/idna_1642433548627/work
imageio==2.19.3
importlib-metadata @ file:///Users/runner/miniforge3/conda-bld/importlib-metadata_1653252883118/work
incremental==21.3.0
install==1.3.5
keras==2.9.0
Keras-Preprocessing @ file:///home/conda/feedstock_root/build_artifacts/keras-preprocessing_1610713559828/work
kiwisolver==1.4.3
libclang==14.0.1
llvmlite==0.38.1
Markdown @ file:///home/conda/feedstock_root/build_artifacts/markdown_1651821407140/work
matplotlib==3.5.2
meshio @ file:///home/conda/feedstock_root/build_artifacts/meshio_1647016219543/work
mpmath @ file:///home/conda/feedstock_root/build_artifacts/mpmath_1612895720168/work
multidict @ file:///Users/runner/miniforge3/conda-bld/multidict_1648882473246/work
netCDF4 @ file:///Users/runner/miniforge3/conda-bld/netcdf4_1630448040847/work
numpy==1.22.4
oauthlib @ file:///home/conda/feedstock_root/build_artifacts/oauthlib_1643507977997/work
opt-einsum @ file:///home/conda/feedstock_root/build_artifacts/opt_einsum_1617859230218/work
packaging==21.3
pandas==1.4.2
paramiko @ file:///home/conda/feedstock_root/build_artifacts/paramiko_1652792654356/work
pexpect @ file:///home/conda/feedstock_root/build_artifacts/pexpect_1602535608087/work
Pillow==9.1.1
plumbum @ file:///home/conda/feedstock_root/build_artifacts/plumbum_1640298612513/work
progressbar2==4.0.0
protobuf==3.20.1
protoc-gen-swagger==0.1.0
psutil==5.9.1
ptyprocess @ file:///home/conda/feedstock_root/build_artifacts/ptyprocess_1609419310487/work/dist/ptyprocess-0.7.0-py2.py3-none-any.whl
pyasn1==0.4.8
pyasn1-modules==0.2.7
pycparser @ file:///home/conda/feedstock_root/build_artifacts/pycparser_1636257122734/work
Pygments @ file:///home/conda/feedstock_root/build_artifacts/pygments_1650904496387/work
pyiges==0.2.1
PyJWT @ file:///home/conda/feedstock_root/build_artifacts/pyjwt_1652398519695/work
PyNaCl @ file:///Users/runner/miniforge3/conda-bld/pynacl_1649520026469/work
pyOpenSSL @ file:///home/conda/feedstock_root/build_artifacts/pyopenssl_1643496850550/work
pyparsing==3.0.9
PySocks @ file:///Users/runner/miniforge3/conda-bld/pysocks_1648857374584/work
python-dateutil @ file:///home/conda/feedstock_root/build_artifacts/python-dateutil_1626286286081/work
python-utils==3.3.3
pytz @ file:///home/conda/feedstock_root/build_artifacts/pytz_1647961439546/work
pyu2f @ file:///home/conda/feedstock_root/build_artifacts/pyu2f_1604248910016/work
pyvista==0.34.1
requests @ file:///home/conda/feedstock_root/build_artifacts/requests_1654796491502/work
requests-oauthlib @ file:///home/conda/feedstock_root/build_artifacts/requests-oauthlib_1643557462909/work
rich @ file:///home/conda/feedstock_root/build_artifacts/rich_1652047427553/work
rsa @ file:///home/conda/feedstock_root/build_artifacts/rsa_1637781155505/work
scipy @ file:///Users/runner/miniforge3/conda-bld/scipy_1653074075583/work
scooby==0.5.12
Send2Trash @ file:///home/conda/feedstock_root/build_artifacts/send2trash_1628511208346/work
six @ file:///home/conda/feedstock_root/build_artifacts/six_1590081179328/work
sympy @ file:///Users/runner/miniforge3/conda-bld/sympy_1647802590691/work
tensorboard==2.9.1
tensorboard-data-server @ file:///Users/runner/miniforge3/conda-bld/tensorboard-data-server_1649932898779/work/tensorboard_data_server-0.6.0-py3-none-macosx_11_0_arm64.whl
tensorboard-plugin-wit @ file:///home/conda/feedstock_root/build_artifacts/tensorboard-plugin-wit_1641458951060/work/tensorboard_plugin_wit-1.8.1-py3-none-any.whl
tensorflow-estimator==2.9.0
tensorflow-macos==2.9.2
tensorflow-metal==0.5.0
termcolor==1.1.0
tqdm==4.64.0
Twisted==22.4.0
txaio==22.2.1
typing-extensions @ file:///home/conda/feedstock_root/build_artifacts/typing_extensions_1602702424206/work
uritemplate==4.1.1
urllib3 @ file:///home/conda/feedstock_root/build_artifacts/urllib3_1647489083693/work
vtk==9.0.3
wcwidth @ file:///home/conda/feedstock_root/build_artifacts/wcwidth_1600965781394/work
webencodings==0.5.1
Werkzeug @ file:///home/conda/feedstock_root/build_artifacts/werkzeug_1651670883478/work
wrapt @ file:///Users/runner/miniforge3/conda-bld/wrapt_1624972047019/work
wslink==1.6.5
yarl @ file:///Users/runner/miniforge3/conda-bld/yarl_1648966561632/work
zipp @ file:///home/conda/feedstock_root/build_artifacts/zipp_1649012893348/work
zope.interface==5.4.0

**Comment by germa89** [comment]
I think you could install ``pymapdl-reader`` version 0.51.7 (released in 24 Nov 2021) using this [link](https://stackoverflow.com/questions/5226311/installing-specific-package-version-with-pip). 

And then install ``pymadpl-core`` using normal ``pip install``.

You 
If you are looking for an older version, you need to:
1. Install the version of ``pymapdl-reader`` you want.
2. Clone the ``pymapdl`` package.
3. Edit the ``pyproject.tml`` file to decrease ``pymapdl-reader`` required version as much as you need. 
4. Then install the cloned ``pymapdl`` using editable flag (``pip install -e .``).

I think this should fit your purpose.

**Comment by kp3393** [comment]
Still the same error. I tried v0.51.5, v0.51.6, and v0.51.7. Still error: legacy-install-failure.
I tried pip (as mentioned above) and installing it directly from the git, I also cleaned my virtual environment and created a new one (to remove old installations), but nothing works.

I appreciate your help, though! Appreciate it.
I will be very grateful if the team can address this issue :).

**Comment by akaszynski** [comment]
We stopped building wheels for MacOS several releases back. If this is still needed we can start making the wheels again. Recommending https://cibuildwheel.readthedocs.io/

@germa89, do you think you'll be able to handle this?

**Comment by germa89** [comment]
> We stopped building wheels for MacOS several releases back. If this is still needed we can start making the wheels again. Recommending https://cibuildwheel.readthedocs.io/
> 
> @germa89, do you think you'll be able to handle this?

@akaszynski  let me give a shot when I get the new laptop.

**Comment by akaszynski** [comment]
@germa89, you can do this without a Mac, just use `cibuildwheel`. See https://github.com/pyvista/pymeshfix/pull/44/files.
